views/js/controller/items/action.js

```diff
@@ -68,8 +68,17 @@ define([
             fullPage: true,
             pluginsOptions: config.pluginsOptions
         }, _.isUndefined);
-
-        previewerFactory(config.provider, config.uri, config.state, previewerConfig);
+        const getProvider = id => {
+            if (!id || !config.providers) {
+                return config.provider
+            }
+            const previewerId = parseInt(`${id}`.split('-').pop(), 10) || 0;
+            if (!config.providers[previewerId]) {
+                return config.provider;
+            }
+            return config.providers[previewerId].id;
+        };
+        previewerFactory(getProvider(this.id) || 'qtiItem', config.uri, config.state, previewerConfig);
```

Comment:
Just my opinion, but maybe it makes more sense to fully delegate the provider discovery to the `getProvider` rather than having a `getProvider(this.id) || 'qtiItem'` here.

Reply:
Doing so will bring back the complexity that is avoided by this inline helper. `|| 'qtiItem'` is made at one place, while putting it inside the helper will require to make it in each early return.

Reply:
I am not sure this is that expensive to be inside the method. IMO encapsulate the responsibility to define which provider to use should be fully inside that function. Anyway, this is a small function only used in this point, so we can keep as preferable :)